process/coordinator/process.go

```diff
@@ -1162,6 +1179,157 @@ func getNumOfCrossShardScCallsOrSpecialTxs(
 	return numCrossShardScCallsOrSpecialTxs
 }
 
+// VerifyCreatedMiniBlocks re-checks gas used and generated fees in the given block
+func (tc *transactionCoordinator) VerifyCreatedMiniBlocks(header data.HeaderHandler, body *block.Body) error {
+	if header.GetEpoch() < tc.blockGasAndFeesReCheckEnableEpoch {
+		return nil
+	}
+
+	mapMiniBlockTypeAllTxs := tc.getAllTransactions(body)
+
+	err := tc.verifyGasLimit(body, mapMiniBlockTypeAllTxs)
+	if err != nil {
+		return err
+	}
+
+	err = tc.verifyFees(header, body, mapMiniBlockTypeAllTxs)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (tc *transactionCoordinator) getAllTransactions(body *block.Body) map[block.Type]map[string]data.TransactionHandler {
+	mapMiniBlockTypeAllTxs := make(map[block.Type]map[string]data.TransactionHandler)
+	for _, miniBlock := range body.MiniBlocks {
+		_, ok := mapMiniBlockTypeAllTxs[miniBlock.Type]
```

Comment:
could there be 2 miniblocks with the same type? Will the second miniblock of the same type be taken into consideration?